crates/bevy_animation_rig/src/skinned_mesh.rs

```diff
@@ -0,0 +1,144 @@
+use bevy_asset::{Assets, Handle, HandleUntyped};
+use bevy_ecs::{
+    entity::{Entity, EntityMap, MapEntities, MapEntitiesError},
+    reflect::{ReflectComponent, ReflectMapEntities},
+    system::{Query, Res, ResMut},
+};
+use bevy_math::Mat4;
+use bevy_pbr::render_graph;
+use bevy_reflect::{Reflect, TypeUuid};
+use bevy_render::{
+    pipeline::PipelineDescriptor,
+    render_graph::{base::node, RenderGraph, RenderResourcesNode},
+    renderer::{
+        RenderResource, RenderResourceHints, RenderResourceIterator, RenderResourceType,
+        RenderResources,
+    },
+    shader::{Shader, ShaderStage},
+    texture::Texture,
+};
+use bevy_transform::components::GlobalTransform;
+
+pub const SKINNED_MESH_PIPELINE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(PipelineDescriptor::TYPE_UUID, 0x14db1922328e7fcc);
+
+#[derive(Debug, Default, Clone, Reflect)]
+#[reflect(Component, MapEntities)]
+pub struct SkinnedMesh {
+    pub bindposes: Handle<SkinnedMeshInverseBindposes>,
+    pub joint_entities: Vec<Entity>,
+    pub joint_transforms: Vec<Mat4>,
+}
+
+impl MapEntities for SkinnedMesh {
+    fn map_entities(&mut self, entity_map: &EntityMap) -> Result<(), MapEntitiesError> {
+        for entity in &mut self.joint_entities {
+            *entity = entity_map.get(*entity)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl RenderResource for SkinnedMesh {
+    fn resource_type(&self) -> Option<RenderResourceType> {
+        Some(RenderResourceType::Buffer)
+    }
+
+    fn write_buffer_bytes(&self, buffer: &mut [u8]) {
+        let transform_size = std::mem::size_of::<[f32; 16]>();
+
+        for (index, transform) in self.joint_transforms.iter().enumerate() {
+            transform.write_buffer_bytes(
+                &mut buffer[index * transform_size..(index + 1) * transform_size],
+            );
+        }
+    }
+
+    fn buffer_byte_len(&self) -> Option<usize> {
+        Some(self.joint_transforms.len() * std::mem::size_of::<[f32; 16]>())
+    }
+
+    fn texture(&self) -> Option<&Handle<Texture>> {
+        None
+    }
+}
+
+impl RenderResources for SkinnedMesh {
+    fn render_resources_len(&self) -> usize {
+        1
+    }
+
+    fn get_render_resource(&self, index: usize) -> Option<&dyn RenderResource> {
+        if index == 0 {
+            Some(self)
+        } else {
+            None
+        }
+    }
+
+    fn get_render_resource_name(&self, index: usize) -> Option<&str> {
+        if index == 0 {
+            Some("JointTransforms")
+        } else {
+            None
+        }
+    }
+
+    fn get_render_resource_hints(&self, _index: usize) -> Option<RenderResourceHints> {
+        Some(RenderResourceHints::BUFFER)
+    }
+
+    fn iter(&self) -> RenderResourceIterator {
+        RenderResourceIterator::new(self)
+    }
+}
+
+#[derive(Debug, TypeUuid)]
+#[uuid = "b9f155a9-54ec-4026-988f-e0a03e99a76f"]
+pub struct SkinnedMeshInverseBindposes(pub Vec<Mat4>);
+
+pub fn skinned_mesh_setup(
```

Comment:
It's not clear to me that these two systems should be pub.

Reply:
I looked at how other bevy crates and they all seem to be public. So I made these 2 systems pub as well.

Reply:
That's perfectly valid logic. If we want to change that we can do so cohesively later.